app/code/Magento/Wishlist/Controller/Index/Update.php

```diff
@@ -88,6 +102,10 @@ public function execute()
                 $qty = null;
                 if (isset($post['qty'][$itemId])) {
                     $qty = $this->quantityProcessor->process($post['qty'][$itemId]);
+                    if (is_double($qty)) {
+                        $stockItem = $this->stockItemRepository->get($itemId);
+                        $qty = !$stockItem->getIsQtyDecimal() ? floor($qty) : $qty;
```

Comment:
I have a concern about floor method call, i think better check qty and return message for user that qty is not valid. The same behavior used in case when you try to update qty to decimal in the shopping cart. You can find logic regarding this check in: https://github.com/magento/magento2/blob/36bfa583f12ccf79c754920d6b6288e67fcc9961/app/code/Magento/CatalogInventory/Model/StockStateProvider.php#L164